app/lib/discourse_automation/scriptable.rb

```diff
@@ -153,6 +153,22 @@ def self.send_pm(pm, sender: Discourse.system_user.username, delay: nil, automat
           end
         end
       end
+
+      def self.close_topic(topic, username: Discourse.system_user.username, message: nil)
+        user = User.find_by_username(username)
+
+        # FIXME: these validations should likely raise instead of silently returning
+        return unless user
+        return unless Guardian.new(user).can_moderate?(topic)
+
+        topic.update_status('closed', true, user)
+
+        unless message.blank?
+          topic_closed_post = topic.posts.where(action_code: 'closed.enabled').last
+          topic_closed_post.raw = message
```

Comment:
is there a min length for these mesages?

Reply:
good catch. Before saving, I could either use `Post#skip_validation` (to skip the rules) or `Post#valid?` (to avoid errors if we know they'll happen).

In the latter case I don't think we want to raise, but there's also no good way to alert the user; the correct time to validate would be when setting up the script, but I don't see any length restriction options in the `text` component.

What do?

Reply:
I think skip_validation might be enough for now. I want to get to some error handling/logging in automations but this will be a fair amount of work.